sdk/dynamicconfig/sdk/metric/controller/push/push.go

```diff
@@ -197,7 +253,27 @@ func (c *Controller) tick() {
 	defer c.processor.Unlock()
 
 	c.processor.StartCollection()
-	c.accumulator.Collect(ctx)
+
+	if c.notifier == nil {
+		c.accumulator.Collect(ctx)
+	} else {
+		// Export all metrics with the same CollectionPeriod at the same time.
+		overdue := []pb.ConfigResponse_MetricConfig_Schedule_CollectionPeriod{}
+		now := c.clock.Now()
+
+		for period, lastCollect := range c.lastCollected {
+			expectedExportTime := lastCollect.Add(time.Duration(period) * time.Second)
+
+			// Check if enough time elapsed since metric with a CollectionPeriod were
+			// last exported.
+			if expectedExportTime.Before(now) || expectedExportTime.Equal(now) {
+				overdue = append(overdue, period)
+				c.lastCollected[period] = now
```

Comment:
Interesting philosophical question about collection periods: if you are supposed to collect every 30 seconds, you collected at time t0, and then you were overdue and collected again at t0+40 seconds, should you do your next collection at t0+60 seconds, or t0+70 seconds? This implementation would do t0+70 seconds which is 30 seconds in the future, but you might also consider t0+60 seconds based on the idea that every 30 seconds on average 1 export is attempted. I would probably lean toward t0+60 seconds as more faithful to the idea of a collection period, but just adding 30 seconds to current time and getting t0+70s is easier to implement and has fewer complications (eg, what happens if you miss a collection period? if you're only a few seconds away from the next collection? etc). Food for thought - let's discuss this a bit offline.